pkg/chartmuseum/server/multitenant/index.go

```diff
@@ -39,49 +39,45 @@ func (server *MultiTenantServer) getIndexFile(log cm_logger.LoggingFn, repo stri
 		return nil, &HTTPError{http.StatusInternalServerError, errStr}
 	}
 
-	fo := <-server.getChartList(log, repo)
+	// if cache is nil, regenerate it
+	if len(entry.RepoIndex.Entries) == 0 {
 
-	if fo.err != nil {
-		errStr := fo.err.Error()
-		log(cm_logger.ErrorLevel, errStr,
-			"repo", repo,
-		)
-		return nil, &HTTPError{http.StatusInternalServerError, errStr}
-	}
-
-	objects := server.getRepoObjectSlice(entry)
-	diff := cm_storage.GetObjectSliceDiff(objects, fo.objects, server.TimestampTolerance)
+		fo := <-server.getChartList(log, repo)
 
-	// return fast if no changes
-	if !diff.Change {
-		log(cm_logger.DebugLevel, "No change detected between cache and storage",
-			"repo", repo,
-		)
-		return entry.RepoIndex, nil
-	}
-
-	log(cm_logger.DebugLevel, "Change detected between cache and storage",
-		"repo", repo,
-	)
-
-	ir := <-server.regenerateRepositoryIndex(log, entry, diff)
-	newRepoIndex := ir.index
-
-	if ir.err != nil {
-		errStr := ir.err.Error()
-		log(cm_logger.ErrorLevel, errStr,
-			"repo", repo,
-		)
-		return newRepoIndex, &HTTPError{http.StatusInternalServerError, errStr}
-	}
+		if fo.err != nil {
+			errStr := fo.err.Error()
+			log(cm_logger.ErrorLevel, errStr,
+				"repo", repo,
+			)
+			return nil, &HTTPError{http.StatusInternalServerError, errStr}
+		}
 
-	if server.UseStatefiles {
-		// Dont wait, save index-cache.yaml to storage in the background.
-		// It is not crucial if this does not succeed, we will just log any errors
-		go server.saveStatefile(log, repo, ir.index.Raw)
+		objects := server.getRepoObjectSlice(entry)
+		diff := cm_storage.GetObjectSliceDiff(objects, fo.objects, server.TimestampTolerance)
```

Comment:
@Ailsa-Wu, @jdolitsky 

Will this method refer to the [latest improved version](https://github.com/chartmuseum/storage/blob/7feb4823f486829bd521ecf6ca31b9863601975e/storage.go#L57)?

Reply:
Just updated to latest storage release so yes 👍